src/amo/commands/getAddon.ts

```diff
@@ -0,0 +1,82 @@
+import * as fs from "fs";
+import * as vscode from "vscode";
+
+import { addonInfoResponse } from "../types";
+import { downloadAddon } from "../utils/addonDownload";
+import { extractAddon } from "../utils/addonExtract";
+import { getAddonInfo } from "../utils/addonInfo";
+import { getVersionChoice } from "../utils/addonVersions";
+
+export async function downloadAndExtract() {
+  const input: string | undefined = await vscode.window.showInputBox({
+    prompt: "Enter Addon Slug, GUID, or URL",
+    title: "Assay",
+  });
+
+  if (!input) {
+    return;
+  }
+
+  // Retrieve version
+  const versionInfo = await getVersionChoice(input);
+  if (!versionInfo) {
+    return;
+  }
```

Comment:
Generally if you're using promises the best practise is to reject (or raise an error if it's an implicit `Promise` with `async`) for failure cases rather than return `undefined`, because that way it cascades through and you don't have to check for undefined after each call.